src/pages/QuizUi/QuizContent.jsx

```diff
@@ -0,0 +1,172 @@
+      setUserPerformance({ correctCount, totalCount });
+    }
+  }, [quizFinished, answersCorrectness, questions.length]);
+
+  const handleAnswerSelected = (answer, resetCallback) => {
+    const isCorrect = answer === questions[currentQuestionIndex].correctOption;
+    setAnswersCorrectness((prev) => {
+      const newAnswers = [...prev];
+      newAnswers[currentQuestionIndex] = { isCorrect };
+      return newAnswers;
+    });
+
+    //changes to the next question if more questions exist
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex((prevIndex) => prevIndex + 1);
+    }
+
+    resetCallback();
+  };
+
+  const handleSkipQuestion = (resetCallback) => {
+    //changes to the next question if more questions exist
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex((prevIndex) => prevIndex + 1);
+    }
+
+    resetCallback();
+  };
+
+  const handleBackQuestion = (resetCallback) => {
+    //changes to the next question if more questions exist
+    if (currentQuestionIndex < questions.length && currentQuestionIndex > 0) {
+      setCurrentQuestionIndex((prevIndex) => prevIndex - 1);
+    }
+
+    resetCallback();
+  };
+
+  return (
+    <>
+      {!quiz ? (
+        <Loading />
+      ) : (
+        <Container sx={containerStyles}>
+          <Typography sx={titleStyles}>{quizTitle}</Typography>
+          <Container
+            style={{
+              display: 'flex',
+              flexDirection: 'row',
+              justifyContent: 'left',
+              paddingInline: '30px',
+              paddingBottom: '20px',
+            }}
+          >
+            <Typography
+              variant="body2"
+              sx={{ color: 'rgb(223, 221, 221)', pl: 2, pr: 2 }}
+            >
+              {currentQuestionIndex < questions.length
+                ? `${currentQuestionIndex + 1} of ${questions.length}`
+                : `${currentQuestionIndex} of ${questions.length}`}
+            </Typography>
+            {questions.map((q, index) => {
+              const correctnessInfo = answersCorrectness[index];
+              const isCurrentQuestion = index === currentQuestionIndex;
+
+              return (
+                <div
+                  key={q.id}
+                  style={{
+                    backgroundColor: correctnessInfo
+                      ? correctnessInfo.isCorrect
+                        ? '#2e7d32'
+                        : '#d32f2f'
+                      : 'rgb(211, 209, 209)',
+                    border: isCurrentQuestion ? '2px solid #1976d2' : 'none',
+                    borderRadius: '50%',
+                    width: 9,
+                    height: 9,
+                    margin: 6,
+                    display: 'inline-block',
+                  }}
+                ></div>
+              );
+            })}
+          </Container>
+          {currentQuestionIndex < questions.length &&
+            (!quizFinished ? (
```

Comment:
When I finish a quiz I only see text on the screen. Can you think of a way to provide the user with options to take action after they finish a quiz? It'd be good to find a way to give users the opportunity to do something else like look at other quizzes.

Reply:
Yes, I actually made a button like this, but I misread the colors in Figma and made the text color the background color so it was illegible on the page!  that is now fixed, thanks again Eli!

Reply:
Ah much better!